### PR TITLE
add component diagram for deployment overview

### PR DESCRIPTION
**Pull request by rianfowler**
# Proposed Changes
Adds this diagram 
Screen Shot 2021-12-08 at 11 45 26 AM

Describe the big picture of your changes here.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

- [ ] My changes generate no new warnings
- [ ] Logging has been added where applicable (Error handling, user-driven events, API requests/responses)

- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] Tests meet the test coverage threshold

## Reviewer Checklist

- [ ] Is the code easy to understand?
- [ ] Are the changes well documented?

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?

- [ ] Should any logging or debugging information be added or removed?

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining how you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -1,5 +1,66 @@
 # Deployment (WIP)
 
+## Deployment components overview 
+
+### lighthouse-embark release
```
Comment (rianfowler) [inline docs/deployment.md:5]:
```suggestion
### lighthouse-embark and lighthosue-embark-deployment
```

File: docs/deployment.md [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
+
+[fdep] --> fev: pushes
+[bdep] --> bev: pushes
+[rel] --> [fdep]: triggers
+[rel] --> [bdep]: triggers
+
+
+```
+
+### lighthouse-embark-deployment deployment
```
Comment (rianfowler) [inline docs/deployment.md:38]:
```suggestion
### Argo deployment
```

File: docs/deployment.md [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+
+
+package "Argo" {
+  component dep as "deployment"
+}
+
+[dep] --> fe: reads
+[dep] --> be: reads
+[dep] --> [fev]: reads
+[dep] --> [bev]: reads
```
Comment (rianfowler) [inline docs/deployment.md:61]:
```suggestion
package "Delivery infrastructure" {
  component argo as "Argo"
  component dep as "Kubernetes"
}

[argo] --> fe: reads
[argo] --> be: reads
[dep] --> [fev]: reads
[dep] --> [bev]: reads
```

File: docs/deployment.md [hunk 4]
```diff
@@ -1,5 +1,67 @@
...
+}
+
+
+package "Delivery infrastructure" {
+  component argo as "Argo"
+  component dep as "Kubernetes"
+}
+
+[argo] --> fe: reads
+[argo] --> be: reads
```
Comment (rianfowler) [inline docs/deployment.md:60]:
```suggestion
[argo] --> be: reads
[argo] --> dep: triggers
```

**Review by keyluck — APPROVED** [review]

**Comment by rianfowler** [comment]
Screen Shot 2021-12-08 at 12 13 46 PM
added this too

**Pull request merged**